crates/evm/evm/src/executors/fuzz/mod.rs

```diff
@@ -150,14 +156,17 @@ impl FuzzedExecutor {
             (call.traces.clone(), call.cheatcodes.map(|c| c.breakpoints))
         };
 
+        // decide whether to use trace logs or only error logs
+        let actual_logs = if show_logs { fuzz_result.logs } else { call.logs };
+
         let mut result = FuzzTestResult {
             first_case: fuzz_result.first_case.unwrap_or_default(),
             gas_by_case: fuzz_result.gas_by_case,
             success: run_result.is_ok(),
             reason: None,
             counterexample: None,
-            decoded_logs: decode_console_logs(&call.logs),
-            logs: call.logs,
+            decoded_logs: decode_console_logs(&actual_logs),
```

Comment:
I think we can get rid of `decode_logs` from `FuzzTestResult` as logs are decoded already when test result created

https://github.com/foundry-rs/foundry/blob/fdf165b65c7344415b684eb4ad8215ac7b713be2/crates/forge/src/result.rs#L493-L515